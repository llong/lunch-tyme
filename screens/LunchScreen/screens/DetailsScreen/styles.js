import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  detailsHeader: {
    backgroundColor: '#34B379',
    height: 60,
    paddingHorizontal: 12,
    justifyContent: 'center'
  },
  restaurantName: {
    color: '#FFFFFF',
    fontSize: 16,
    fontFamily: 'Avenir Next Demi Bold'
  },
  restaurantCategory: {
    color: '#FFFFFF',
    fontSize: 12
  },
  mapContainer: {
    height: 180,
    backgroundColor: 'red'
  },
  detailsContainer: {
    paddingVertical: 16,
    paddingHorizontal: 12,
    backgroundColor: '#FFFFFF'
  },
  detailsText: {
    fontSize: 16,
    marginBottom: 26,
    lineHeight: 24
  }
});
