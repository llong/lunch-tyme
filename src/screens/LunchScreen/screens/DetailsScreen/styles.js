import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  detailsHeader: {
    backgroundColor: '#34B379',
    height: 60,
    paddingHorizontal: 12,
    justifyContent: 'center',
  },
  headerIcon: {
    height: 24,
    resizeMode: 'contain',
  },
  restaurantName: {
    color: '#FFFFFF',
    fontSize: 16,
    fontWeight: '600',
    fontFamily: 'Avenir Next Demi Bold',
  },
  restaurantCategory: {
    color: '#FFFFFF',
    fontSize: 12,
  },
  mapContainer: {
    height: 180,
    backgroundColor: '#E6E6E6',
  },
  detailsContainer: {
    paddingVertical: 16,
    paddingHorizontal: 12,
    backgroundColor: '#FFFFFF',
  },
  detailsText: {
    fontSize: 16,
    marginBottom: 26,
    lineHeight: 24,
  },
});
