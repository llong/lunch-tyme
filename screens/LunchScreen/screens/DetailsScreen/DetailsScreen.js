import React from 'react';
import { View, Text, Image } from 'react-native';
import styles from './styles';
import MapView , { Marker } from 'react-native-maps';
import { connect } from 'react-redux';
import MapIcon from '../../../../assets/icons/icon_map.png';

class DetailsView extends React.Component {
  static navigationOptions = ({navigation}) => {
    return {
      title: 'Lunch Tyme',
      headerRight: (
        <Image source={ MapIcon } style={ { marginRight: 12 } } />
      )
    };
  }
  render() {
    const { activeRestaurant: { name, category, location, contact } } = this.props;
    return (
      <View style={ { flex: 1 } }>
        <View style={ styles.mapContainer }>
          <MapView
            style={{flex:1, alignSelf: 'stretch'}}
            initialRegion={{
              latitude: location.lat,
              longitude: location.lng,
              latitudeDelta: 0.0522,
              longitudeDelta: 0.0421,
            }}
          >
             <Marker
              coordinate={ {latitude: location.lat, longitude: location.lng } }
              title={ name }
              description={ category }
            />
          </MapView>
        </View>
        <View style={ styles.detailsHeader }>
          <Text style={ styles.restaurantName } >{ name }</Text>
          <Text style={ styles.restaurantCategory } >{ category }</Text>
        </View>
        <View style={ styles.detailsContainer }>
          { location && location.formattedAddress && (
            <Text style={ styles.detailsText }>
              {`${location.formattedAddress[0]}\n${location.formattedAddress[1]}`}
              </Text>
            )
          }
          { contact && contact.formattedPhone && <Text style={ styles.detailsText }>{contact.formattedPhone }</Text> }
          { contact && contact.twitter && <Text style={ styles.detailsText }>{`@${ contact.twitter }`}</Text> }
        </View>
      </View>
    )
  }
}

function mapStateToProps(state) {
  return {
    activeRestaurant: state.activeRestaurant
  }
}

export default connect(mapStateToProps)(DetailsView);
