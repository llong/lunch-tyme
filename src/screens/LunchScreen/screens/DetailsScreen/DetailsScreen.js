// @flow

import React from 'react';
import { View, Text, Image } from 'react-native';
import MapView, { Marker } from 'react-native-maps';
import { connect } from 'react-redux';
import Mapbox from '@mapbox/react-native-mapbox-gl';
import styles from './styles';
import MapIcon from '../../../../assets/icons/icon_map.png';

type Props = {
  activeRestaurant: {
    name: String,
    category: String,
    location: Object,
  }
}

Mapbox.setAccessToken('pk.eyJ1IjoiaXp0ZWwiLCJhIjoiY2p2Z2ozZ3FzMDdsNDRhcDc2YWw4ZG96aCJ9.6VHDT7SBmiGl5O6-feEeeg');

export class DetailsView extends React.Component<Props> {
  static navigationOptions = {
    title: 'Lunch Tyme',
    headerRight: (
      <Image source={MapIcon} style={styles.headerIcon} />
    ),
  }

  render() {
    const {
      activeRestaurant: {
        name,
        category,
        location,
        contact,
      },
    } = this.props;
    return (
      <View style={{ flex: 1 }}>
        <View style={styles.mapContainer}>
          <Mapbox.MapView
            style={{ flex: 1, alignSelf: 'stretch' }}
            centerCoordinate={[location.lng, location.lat]}
            zoomLevel={12}
          >
            <Mapbox.PointAnnotation
              id={name}
              coordinate={[location.lng, location.lat]}
              title={name}
              snippet={category}
              selected
            >
              <Mapbox.Callout title={name} />
            </Mapbox.PointAnnotation>
          </Mapbox.MapView>
        </View>
        <View style={styles.detailsHeader}>
          <Text style={styles.restaurantName}>{ name }</Text>
          <Text style={styles.restaurantCategory}>{ category }</Text>
        </View>
        <View style={styles.detailsContainer}>
          { location && location.formattedAddress && (
            <Text style={styles.detailsText}>
              {`${location.formattedAddress[0]}\n${location.formattedAddress[1]}`}
            </Text>
          )}
          { contact && contact.formattedPhone && (
            <Text style={styles.detailsText}>{contact.formattedPhone }</Text>
          )}
          { contact && contact.twitter && (
            <Text style={styles.detailsText}>{`@${contact.twitter}`}</Text>
          )}
        </View>
      </View>
    );
  }
}

function mapStateToProps(state) {
  return {
    activeRestaurant: state.activeRestaurant,
  };
}

export default connect(mapStateToProps)(DetailsView);
