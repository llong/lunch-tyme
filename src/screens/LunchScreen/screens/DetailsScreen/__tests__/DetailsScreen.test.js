import React from 'react';
import { shallow } from 'enzyme';
import { DetailsView } from '../DetailsScreen';

const props = {
  activeRestaurant: {
    name: 'McDonalds',
    category: 'Fast Food',
    location: {
      lat: 0.123,
      lng: 0.321,
    },
  },
};

describe('DetailsScreen', () => {
  const wrapper = shallow(<DetailsView {...props} />, { disableLifecycleMethods: true });
  it('should render component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
