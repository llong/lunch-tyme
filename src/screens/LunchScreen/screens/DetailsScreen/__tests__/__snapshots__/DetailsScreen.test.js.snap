// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsScreen should render component 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "red",
        "height": 180,
      }
    }
  >
    <MapView
      initialRegion={
        Object {
          "latitude": 0.123,
          "latitudeDelta": 0.0522,
          "longitude": 0.321,
          "longitudeDelta": 0.0421,
        }
      }
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
        }
      }
    >
      <MapMarker
        coordinate={
          Object {
            "latitude": 0.123,
            "longitude": 0.321,
          }
        }
        description="Fast Food"
        stopPropagation={false}
        title="McDonalds"
      />
    </MapView>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#34B379",
        "height": 60,
        "justifyContent": "center",
        "paddingHorizontal": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Avenir Next Demi Bold",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      McDonalds
    </Text>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 12,
        }
      }
    >
      Fast Food
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "paddingHorizontal": 12,
        "paddingVertical": 16,
      }
    }
  />
</View>
`;
